File: server/src/halfpipe/Farbe.scala
package halfpipe

sealed abstract class Farbe(val wert: Int)

case object Karo extends Farbe(9)
case object Herz extends Farbe(10)
case object Pik extends Farbe(11)
case object Kreuz extends Farbe(12)

object Farbe {
  def alle = List(Karo, Herz, Pik, Kreuz)
}

File: core/src/halfpipe/Farbe.java
package halfpipe;

import java.util.NoSuchElementException;

public enum Farbe {
  KARO("Karo", 9),
  HERZ("Herz", 10),
  PIK("Pik", 11),
  KREUZ("Kreuz", 12),
  NULL("Null", 0);

  private Farbe(String name, int wert) {
    this.name = name;
    this.wert = wert;
  }

  public final String name;
  public final int wert;

  public static Farbe fromName(String name) {
    switch (name) {
      case "Karo": return KARO;
      case "Herz": return HERZ;
      case "Pik": return PIK;
      case "Kreuz": return KREUZ;
      case "Null": return NULL;
      default: throw new NoSuchElementException("Ungültige Farbe");
    }
  }

}